From 6fed0334c99d3c088752b462d106a84266fb1114 Mon Sep 17 00:00:00 2001
From: Olivier Fourdan <ofourdan@redhat.com>
Date: Wed, 10 Apr 2019 11:01:31 +0200
Subject: [PATCH libICE 1/3] IceListenForWellKnownConnections: Fix memleak

The function `_IceTransMakeAllCOTSServerListeners` allocates memory for
`transConns` which is leaked in case of error.

Signed-off-by: Olivier Fourdan <ofourdan@redhat.com>
---
 src/listenwk.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/listenwk.c b/src/listenwk.c
index 7517ea8..9ff26da 100644
--- a/src/listenwk.c
+++ b/src/listenwk.c
@@ -61,6 +61,7 @@ IceListenForWellKnownConnections (
         strncpy (errorStringRet,
 	    "Cannot establish any listening sockets", errorLength);
 
+	free (transConns);
 	return (0);
     }
 
-- 
2.21.0

